Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Player;
+using UnityEngine;
+
+public class Chase : State
+{
+    public Chase()
+    {
+        name = "Chase";
+    }
+
+    public override void Move(GameObject player, GameObject npc)
+    {
+        Chasing(player, npc);
+    }
+    
+    private void Chasing(GameObject player, GameObject npc)
+    {
+        Vector3 futureLocation = player.transform.position + player.GetComponent<PlayerDataManager>().PlayerData.Velocity * NPCsGlobalVariables.ChasePredictionMultiplier;
```

Comment:
Avoid calling GetComponent each time this function is called
Instead create a private PlayerDataManager variable in the class (or the base class, up to you) like this : 
```c#
    private PlayerDataManager _manager = player.GetComponent<PlayerDataManager>();
```

This way, the call will be done only once at the initialization, and will be more efficient
(do this for all the GetComponents call, a GetComponent call is expensive and will always return the same result, no need to call it everytime we need the same component)

Reply:
Yeah, I forgot to move them into variables when I was cleaning the code. Will do!